framework/include/utils/MooseTypes.h

```diff
@@ -924,6 +925,21 @@ struct enable_bitmask_operators<Moose::RelationshipManagerType>
   static const bool enable = true;
 };
 
+/**
+ * This Maco is used to generate forward declarations for to_json methods for each of the
+ * string-derived class.
+ */
+#define DerivativeStringToJSONDecl(TheName)                                                        \
+  namespace nlohmann                                                                               \
+  {                                                                                                \
+  template <>                                                                                      \
+  struct adl_serializer<TheName>                                                                   \
+  {                                                                                                \
+    static void to_json(json & j, const TheName & v);                                              \
```

Comment:
https://github.com/hugary1995/moose/blob/8428f040c912cb1b59511b3a0ec6e7c51057d8af/framework/src/utils/MooseTypes.C#L33 ?

Reply:
oh, I see it now. this diff is confusing me